src/formatting_stack/util/diff.clj

```diff
@@ -0,0 +1,46 @@
+(ns formatting-stack.util.diff
+  (:require
+   [clojure.java.data :refer [from-java]]
+   [clojure.java.io :as io]
+   [clojure.spec.alpha :as spec]
+   [clojure.string :as string]
+   [formatting-stack.util :refer [rcomp]]
+   [nedap.speced.def :as speced]
+   [nedap.utils.spec.predicates :refer [present-string?]])
+  (:import
+   (io.reflectoring.diffparser.api UnifiedDiffParser)))
+
+(spec/def ::begin pos-int?)
+(spec/def ::end pos-int?)
+(spec/def ::filename present-string?)
+
+(spec/def ::line-numbers
+  (spec/coll-of (spec/keys :req-un [::begin ::end ::filename])))
+
+(speced/defn ^::line-numbers diff->line-numbers
+  "Returns maps of consecutive removals/changes (as `:begin` to `:end`) per `:filename` in `diff`."
```

Comment:
The docstring is quite spec-like

Most of the info can be elided and/or moved to a def-with-doc elsewhere